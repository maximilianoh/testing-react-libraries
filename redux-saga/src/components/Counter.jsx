import { useDispatch, useSelector } from 'react-redux';
import { counterGroup } from '../ducks/';

function App() {
  const dispatch = useDispatch();
  const counter = useSelector(counterGroup.selectors.getCount);
  return (
    <div> 
        <h1> Counter </h1>
        <h1> {counter} </h1>
        <button onClick={() => dispatch(counterGroup.actions.increase())}> Increase </button>
        <button onClick={() => dispatch(counterGroup.actions.decrease())}> Decrease </button>
        <button onClick={() => dispatch(counterGroup.actions.reset())}> Reset </button>
    </div>
  );
}

export default App;