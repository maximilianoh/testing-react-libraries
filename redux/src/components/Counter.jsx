import React, {PureComponent} from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import actions from '../ducks/counter/actions';

class Counter extends PureComponent {
  render() {
    return (
      <div> 
        <h1> Counter </h1>
        <h1> {this.props.count} </h1>
        <button onClick={this.props.handleIncrease}> Increase </button>
        <button onClick={this.props.handleDecrease}> Decrease </button>
        <button onClick={this.props.handleReset}> Reset </button>
      </div>
    )
  }
}

function mapStateToProps(state) {
  return {
    count: state.counterGroup.count,
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({
    handleIncrease: actions.increase,
    handleDecrease: actions.decrease,
    handleReset: actions.reset,
  }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Counter);