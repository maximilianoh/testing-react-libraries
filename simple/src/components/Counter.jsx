function Counter({counter, setCounter, setHistory}) {
  return (
    <div> 
        <h1> Counter </h1>
        <h1> {counter} </h1>
        <button onClick={() => {
          const c = counter+1;
          setCounter( c );
          setHistory(state => ([c, ...state]));}}> 
            Increase 
        </button>
        <button onClick={() => {
          const c = counter-1;
          setCounter( c );
          setHistory(state => ([c, ...state]));}}> 
            Decrease 
        </button>
        <button onClick={() => {
          setCounter(0);
          setHistory([]);}}> 
            Reset 
        </button>
    </div>
  );
}

export default Counter;