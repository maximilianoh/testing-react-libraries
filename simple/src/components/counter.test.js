import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import Counter from './Counter';

test('Default Chip', () => {
    let counter = 0;
    let setCounter = () => {

    };
    let setHistory = () => {
        
    };
    const component = renderer.create(
        <Counter
        counter={counter}
        setCounter={setCounter}
        setHistory={setHistory}
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});


test('Default Chip', () => {
    let counter = 0;
    let setCounter = (a) => {
        counter = a;
    };
    let setHistory = () => {
        
    };
    const { rerender, baseElement } = render(<Counter
        counter={counter}
        setCounter={setCounter}
        setHistory={setHistory}
        />
    );
    expect(counter).toBe(0);
    fireEvent.click(screen.getByText('Increase'));
    expect(baseElement).toMatchSnapshot();
    expect(counter).toBe(1);
    rerender(<Counter
        counter={counter}
        setCounter={setCounter}
        setHistory={setHistory}
        />
    );
    expect(baseElement).toMatchSnapshot();
    expect(counter).toBe(1);
    fireEvent.click(screen.getByText('Increase'));
    expect(counter).toBe(2);
});