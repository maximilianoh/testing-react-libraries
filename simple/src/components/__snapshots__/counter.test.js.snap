// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Chip 1`] = `
<div>
  <h1>
     Counter 
  </h1>
  <h1>
     
    0
     
  </h1>
  <button
    onClick={[Function]}
  >
    Increase
  </button>
  <button
    onClick={[Function]}
  >
    Decrease
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;

exports[`Default Chip 2`] = `
<body>
  <div>
    <div>
      <h1>
         Counter 
      </h1>
      <h1>
         
        0
         
      </h1>
      <button>
        Increase
      </button>
      <button>
        Decrease
      </button>
      <button>
        Reset
      </button>
    </div>
  </div>
</body>
`;

exports[`Default Chip 3`] = `
<body>
  <div>
    <div>
      <h1>
         Counter 
      </h1>
      <h1>
         
        1
         
      </h1>
      <button>
        Increase
      </button>
      <button>
        Decrease
      </button>
      <button>
        Reset
      </button>
    </div>
  </div>
</body>
`;
